### PR TITLE
add rails controller testing

### PR DESCRIPTION
**Pull request by X-Elie-X**
### in this PR I implemented the following:
- Add RSpec as a gem to your project.
- Create a Request spec file for all your controllers.
- Make sure that for each action you check:
If the response status was correct.
If a correct template was rendered.
If the response body includes the correct placeholder text.

### CODE REVIEW

File: spec/Request/posts_specs.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+require 'rails_helper'
```
Comment (gemmen29) [inline spec/Request/posts_specs.rb]: - [ ] Kindly rename the file from `posts_specs.rb` to `posts_spec.rb` as this is the naming convention for test files in order to run tests inside that file.

File: spec/Request/user_specs.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+require 'rails_helper'
+
```
Comment (gemmen29) [inline spec/Request/user_specs.rb]: - [ ] Kindly rename the file from `user_specs.rb` to `user_spec.rb` as this is the naming convention for test files in order to run tests inside that file.

File: spec/Request/user_specs.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+RSpec.describe 'Users', type: :request do
```
Comment (gemmen29) [inline spec/Request/user_specs.rb]: - [ ] Kindly fix the failed test cases
image

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi @X-Elie-X, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/Request/user_spec.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
+require 'rails_helper'
+
+RSpec.describe 'User', type: :request do
```
Comment (OLIPLICHE) [inline spec/Request/user_spec.rb:3]: - [ ] Congrat 👍 

Capture

**Review by OLIPLICHE — APPROVED** [review]
Hi @X-Elie-X, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Highlights
- Implement the changes 👏
- Wrote the tests for controllers👏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**